checkov/common/output/sarif.py

```diff
@@ -0,0 +1,242 @@
+from __future__ import annotations
+
+import itertools
+import json
+from typing import TYPE_CHECKING, Any
+
+from checkov.common.models.enums import CheckResult
+from checkov.common.output.cyclonedx_consts import SCA_CHECKTYPES
+from checkov.version import version
+
+if TYPE_CHECKING:
+    from checkov.common.output.record import Record
+    from checkov.common.output.report import Report
+
+SEVERITY_TO_SARIF_LEVEL = {
+    "critical": "error",
+    "high": "error",
+    "medium": "warning",
+    "low": "note",
+    "none": "none",
+}
+
+
+class Sarif:
+    def __init__(self, reports: list[Report], tool: str | None) -> None:
+        self.reports = reports
+        self.rule_index_map: "dict[str, int]" = {}
+        self.tool = tool if tool else "Bridgecrew"
+
+        self.json = self.create_json()
+
+    def create_json(self) -> dict[str, Any]:
+        return {
+            "$schema": "https://raw.githubusercontent.com/oasis-tcs/sarif-spec/master/Schemata/sarif-schema-2.1.0.json",
+            "version": "2.1.0",
+            "runs": self._create_runs(),
+        }
+
+    def _create_runs(self) -> list[dict[str, Any]]:
+        information_uri = "https://docs.bridgecrew.io" if self.tool.lower() == "bridgecrew" else "https://checkov.io"
+        rules = self._create_rules()  # needs to be invoked before _create_results()
+        results = self._create_results()
+
+        return [
+            {
+                "tool": {
+                    "driver": {
+                        "name": self.tool,
+                        "version": version,
+                        "informationUri": information_uri,
+                        "rules": rules,
+                        "organization": "bridgecrew",
+                    }
+                },
+                "results": results,
+            }
+        ]
+
+    def _create_rules(self) -> list[dict[str, Any]]:
+        rule_idx = 0
+        rules: "list[dict[str, Any]]" = []
```

Comment:
Is quoted typing necessary even if you imported annotations from __future__?

Reply:
it is a bit faster to put into a string literal, I somehow got used to it, when I add complex type hints. But you are right it is not needed when using the `__future__` import

Reply:
I dug the rabbit hole a bit further and found out that `"list[dict[str, Any]]" != list[dict[str, Any]]`
It seems to have no real effect on linting for now but may be worth changing it regardless (?)
https://stackoverflow.com/a/46458497/11918269

Reply:
and as the comment to the answer suggests, don't do that and use the proper function `typing.get_type_hints()`
```python
>>> typing.get_type_hints(bad_foo) == typing.get_type_hints(good_foo)
True
```
😉

Reply:
🤯